src/Tpay/Provider/ValidTpayChannelListProvider.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+declare(strict_types=1);
+
+namespace CommerceWeavers\SyliusTpayPlugin\Tpay\Provider;
+
+use App\Repository\PaymentMethodRepositoryInterface;
+use CommerceWeavers\SyliusTpayPlugin\Model\GatewayConfigInterface;
+use CommerceWeavers\SyliusTpayPlugin\Payum\Exception\UnableToGetBankListException;
+use CommerceWeavers\SyliusTpayPlugin\Tpay\PayGroup;
+use CommerceWeavers\SyliusTpayPlugin\Tpay\PaymentType;
+use Payum\Core\Security\CypherInterface;
+use Sylius\Component\Channel\Context\ChannelContextInterface;
+use Sylius\Component\Core\Model\PaymentMethodInterface;
+use Webmozart\Assert\Assert;
+
+final class ValidTpayChannelListProvider implements ValidTpayChannelListProviderInterface
+{
+    public function __construct(
+        private readonly AvailableTpayChannelListProviderInterface $availableTpayApiBankListProvider,
+        private readonly PaymentMethodRepositoryInterface $paymentMethodRepository,
+        private readonly ChannelContextInterface $channelContext,
+        private readonly CypherInterface $cypher,
+    ) {
+    }
+
+    public function provide(): array
+    {
+        $availableChannels = $this->availableTpayApiBankListProvider->provide();
+
+        /** @var PaymentMethodInterface[] $paymentMethods */
+        $paymentMethods = $this->paymentMethodRepository->findByChannelAndGatewayConfigNameWithGatewayConfig(
+            $this->channelContext->getChannel(),
+            'tpay',
+        );
+
+        Assert::notEmpty($paymentMethods, 'There is no payment method of Tpay type available');
+
+        $paymentMethodsToRemoveByGroupId = [];
+        $hasPblPaymentAvailable = false;
+        foreach ($paymentMethods as $paymentMethod) {
+            /** @var GatewayConfigInterface|null $tpayGatewayConfig */
+            $tpayGatewayConfig = $paymentMethod->getGatewayConfig();
+
+            if (null === $tpayGatewayConfig) {
+                continue;
+            }
+
+            $tpayGatewayConfig->decrypt($this->cypher);
+            $config = $tpayGatewayConfig->getConfig();
+
+            if (!array_key_exists('type', $config)) {
```

Comment:
This is not unit tested

Reply:
In this test, there is still type provided https://github.com/CommerceWeavers/SyliusTpayPlugin/pull/140/files#diff-7a1d71f37f8fe786106bc96fef164a5fd39f598e88c9c38f44062c760d4e4f3aR214

I think we should test the case where the `$config` array is empty for example.

Reply:
Oh ok, missed that https://github.com/CommerceWeavers/SyliusTpayPlugin/pull/140/files#diff-7a1d71f37f8fe786106bc96fef164a5fd39f598e88c9c38f44062c760d4e4f3aR223